### PR TITLE
Bugfix/59 activity by type

### LINKED ISSUE #59: Activity by type, past year is broken
data is wrong, too many columns

image

### PR DESCRIPTION
**Pull request by Lkleven**
should fix https://github.com/Creuna-Oslo/dashboard/issues/59

### CODE REVIEW

File: app/source/components/stats-table/stats-table.jsx [hunk 1]
```diff
@@ -68,13 +67,13 @@ class StatsTable extends React.Component {
       {months &&
         this.lastTwelveMonths().map((month, index) => (
           <React.Fragment key={month}>
-            {index === currentMonth + 1 && <th className="spacer" />}
+            {index === 11 - currentMonth && <th className="spacer" />}
```
Comment (asbjornh) [inline app/source/components/stats-table/stats-table.jsx]: I have no clue what this index logic does. `index === 11 - currentMonth` should probably be extracted to some function with an understandable name. That way the logic also won't have to be duplicated, reducing the chance for bugs in the future

**Review by asbjornh — CHANGES_REQUESTED** [review]

File: app/source/components/stats-table/stats-table.jsx (hunk 1 above)
Comment (Lkleven) [reply app/source/components/stats-table/stats-table.jsx]: I agree, I have extracted this logic into some functions attempting to make things clearer

File: app/source/components/stats-table/stats-table.jsx [hunk 2]
```diff
@@ -62,21 +61,22 @@ class StatsTable extends React.Component {
     return 'table-0';
   };
 
+  firstMonthCurrentYear = index => index === months.length - currentMonth - 1;
```
Comment (asbjornh) [inline app/source/components/stats-table/stats-table.jsx]: Sjekker denne funksjonen om indexen er første måned i nåværende år? I såfall burde den kanskje hete `isFirstMonthOfCurrentYear`?

**Review by asbjornh — CHANGES_REQUESTED** [review]

File: app/source/components/stats-table/stats-table.jsx (hunk 2 above)
Comment (Lkleven) [reply app/source/components/stats-table/stats-table.jsx]: Korrekt! Ja det er vel et bedre navn :)

**Review by asbjornh — APPROVED** [review]

**Pull request merged**